### PR TITLE
Improve docstrings in printer.py

### PR DESCRIPTION
**Pull request by eric-wieser**
The docstring for GaLatexPrinter was really a description of the module in general.
This docstring has been moved to the module docstrnig, and new docstrings written for `GaLatexPrinter` and `GaStrPrinter`.

https://galgebra--420.org.readthedocs.build/en/420/generated/galgebra.printer.html#module-galgebra.printer

### CODE REVIEW

File: galgebra/printer.py [hunk 1]
```diff
@@ -1,5 +1,80 @@
...
+.. note::
+
+    :mod:`galgebra` works out of the box with the usual
+    :ref:`sympy printing <sympy:tutorial-printing>` and will show Latex in
+    IPython by default. In many cases, all that is needed is::
+
+        sympy.init_printing(
+            use_latex='mathjax',
+            latex_printer=galgebra.printer.latex,
+            # described below
```
Comment (utensil) [inline galgebra/printer.py:16]:
```suggestion
            # described below in `galgebra.printer.GaLatexPrinter`
```

File: galgebra/printer.py [hunk 2]
```diff
@@ -1,5 +1,80 @@
...
+        )
+
+    The rest of this section primarily describes an orthogonal feature for
+    writing out ``.tex`` files with :func:`print`.
+
+The latex printer is turned on with the :func:`Format` function
+
+.. function:: Format(Fmode=True, Dmode=True, ipy=False)
+
+where Fmode is the function printing mode that surpresses printing arguments,
```
Comment (utensil) [inline galgebra/printer.py:28]:
```suggestion
where ``Fmode`` is the function printing mode that suppresses printing arguments,
```

File: galgebra/printer.py [hunk 3]
```diff
@@ -1,5 +1,80 @@
...
+Dmode is the derivative printing mode that does not use fractions, and
```
Comment (utensil) [inline galgebra/printer.py:29]:
```suggestion
``Dmode`` is the derivative printing mode that does not use fractions, and
```

File: galgebra/printer.py [hunk 4]
```diff
@@ -1,5 +1,80 @@
...
+ipy=True is the Ipython notebook mode that does not redirect the print output.
```
Comment (utensil) [inline galgebra/printer.py:30]:
```suggestion
``ipy=True`` is the IPython notebook mode that does not redirect the print output.
```

File: galgebra/printer.py [hunk 5]
```diff
@@ -1,5 +1,80 @@
...
+
+The latex output is post processed and displayed with the function
+
+.. function:: xpdf(filename='tmplatex.tex', debug=False)
+
+where filename is the name of the tex file one would keep for future
```
Comment (utensil) [inline galgebra/printer.py:36]:
```suggestion
where ``filename`` is the name of the tex file one would keep for future
```

File: galgebra/printer.py [hunk 6]
```diff
@@ -1,5 +1,80 @@
...
+inclusion in documents and debug=True would display the tex file
```
Comment (utensil) [inline galgebra/printer.py:37]:
```suggestion
inclusion in documents and ``debug=True`` would display the tex file
```

File: galgebra/printer.py [hunk 7]
```diff
@@ -1,5 +1,80 @@
...
+immediately.
+
+There are three options for printing multivectors in latex.  They are
+acessed with the multivector member function
+
+.. function:: galgebra.mv.Mv.Fmt(self, fmt=1, title=None)
+
+where fmt=1, 2, or 3 determines whether the entire multivector A is
```
Comment (utensil) [inline galgebra/printer.py:45]:
```suggestion
where ``fmt``=1, 2, or 3 determines whether the entire multivector A is
```

File: galgebra/printer.py [hunk 8]
```diff
@@ -1,5 +1,80 @@
...
+printed entirely on one line, or one grade is printed per line, or
+one base is printed per line.  If title is not None then the latex
```
Comment (utensil) [inline galgebra/printer.py:47]: The doc for the enum could be changed to bullet points or better, a table.

File: galgebra/printer.py [hunk 9]
```diff
@@ -1,5 +1,80 @@
...
+everthing in the title processing still applies except that the multivector
+formatting is one multivector per line.
+
+For print statements of the form::
+
+    print(title)
+
+where no program variables are printed if title contains `#` then title
+is printed as regular latex line.  If title does not contain `#` then
+title is printed in equation mode. `%` has the same effect in title as
```
Comment (utensil) [inline galgebra/printer.py:76]: Should we merge the `#` `%` doc and code in one place later?

File: galgebra/printer.py [hunk 10]
```diff
@@ -724,12 +788,24 @@ def latex(expr, **settings):
             return s
 
 
-def latex(expr, **settings):
+def latex(expr, **settings) -> str:
+    """
+    Get the latex representation of expr using :class:`GaLatexPrinter`.
+
+    Takes the same options as :func:`sympy.latex`; see that function for
```
Comment (utensil) [inline galgebra/printer.py:795]: We should cross link to sympy for this

Comment (utensil) [reply galgebra/printer.py:795]: As introduced in #328

Comment (utensil) [reply galgebra/printer.py:795]:
```suggestion
    Takes the same options as :py:func:`sympy.latex`; see that function for
```

Comment (utensil) [reply galgebra/printer.py:795]: Maybe this would work.

File: galgebra/printer.py [hunk 11]
```diff
@@ -724,12 +788,24 @@ def latex(expr, **settings):
...
+    more information.
+
+    This can be used as the ``latex_printer`` argument to
+    :func:`sympy.init_printing`, to make IPython always use
```
Comment (utensil) [inline galgebra/printer.py:799]: Same cross link issue here

File: galgebra/printer.py (hunk 10 above)
Comment (eric-wieser) [reply galgebra/printer.py:795]: No, the cause is that I need the fully-qualified name not the alias.

File: galgebra/printer.py (hunk 8 above)
Comment (eric-wieser) [reply galgebra/printer.py:47]: This is just moved from the old docs, don't want to bite off to much here.

File: galgebra/printer.py (hunk 9 above)
Comment (eric-wieser) [reply galgebra/printer.py:76]: Yeah, probably

**Comment by eric-wieser** [comment]
Cross-references and code formatting hopefully all fixed.

**Pull request merged**